File: sdk/typescript/src/documents.ts
/** Document text extraction for catalog chunks (Rust-backed). */

import {
  extractDocumentTextNative,
  extractJsonCatalogDocumentNative,
  extractLevelInfoNative,
  extractMdCatalogDocumentNative,
} from "./native.js";
import type { JsonRecord } from "./types.js";

export function extractJsonCatalogDocument(item: JsonRecord): string | null {
  return extractJsonCatalogDocumentNative(item);
}

export function extractMdCatalogDocument(item: JsonRecord): string | null {
  return extractMdCatalogDocumentNative(item);
}

export function extractDocumentText(itemContent: JsonRecord): string | null {
  return extractDocumentTextNative(itemContent);
}

export function extractLevelInfo(data: JsonRecord): string[] {
  return extractLevelInfoNative(data);
}

File: sdk/go/documents.go
package cytindexer

// ExtractDocumentText extracts plain text from a catalog document item JSON.
func ExtractDocumentText(itemJSON string) (string, error) {
	return cgoExtractDocumentText(itemJSON)
}

// ExtractLevelInfo extracts heading level metadata from a catalog document item.
func ExtractLevelInfo(itemJSON string) (string, error) {
	return cgoExtractLevelInfo(itemJSON)
}

// ExtractJSONCatalogDocument extracts JSON catalog document fields.
func ExtractJSONCatalogDocument(itemJSON string) (string, error) {
	return cgoExtractJSONCatalogDocument(itemJSON)
}

// ExtractMdCatalogDocument extracts markdown catalog document fields.
func ExtractMdCatalogDocument(itemJSON string) (string, error) {
	return cgoExtractMdCatalogDocument(itemJSON)
}
